### PR TITLE
Feature/backend speedup berkeleydb

### PR DESCRIPTION
**Pull request by ninewise**
Working with BerkeleyDB on #422.

_[Original pull request](https://github.ugent.be/unipept/unipept/pull/424) by @ninewise on Tue Oct 07 2014 at 15:36._
_Merged by @bmesuere on Sun Jun 14 2015 at 15:53._

### CODE REVIEW

**Comment by ninewise** [comment]
This runs in about 8 minutes using 500MB cache for the swissprot xml file.


_[Original comment](https://github.ugent.be/unipept/unipept/pull/424#issuecomment-11298) by @ninewise on Tue Oct 07 2014 at 15:50._

File: backend/java/src/main/java/org/unipept/storage/PeptideLoaderData.java [hunk 1]
```diff
@@ -20,84 +23,109 @@
...
-            addGORef = connection
-                    .prepareStatement("INSERT INTO go_cross_references (`uniprot_entry_id`, `go_id`) VALUES (?,?)");
-            addECRef = connection
-                    .prepareStatement("INSERT INTO ec_cross_references (`uniprot_entry_id`, `ec_id`) VALUES (?,?)");
-            lineageExists = connection
-                    .prepareStatement("SELECT COUNT(*) AS aantal FROM lineages WHERE `taxon_id` = ?");
-            getTaxon = connection
-                    .prepareStatement("SELECT rank, parent_id, valid_taxon FROM taxons WHERE id = ?");
-        } catch (SQLException e) {
+            uniprotEntries = new CSVWriter("uniprotEntries.tsv", "uniprot_accession_number", "version", "taxon_id", "type", "protein");
```
Comment (bmesuere) [inline backend/java/src/main/java/org/unipept/storage/PeptideLoaderData.java]: this file should be named `uniprot_entries.tsv`


_[Original comment](https://github.ugent.be/unipept/unipept/pull/424#discussion_r2074) by @bmesuere on Fri Jan 16 2015 at 14:34._

**Comment by bmesuere** [comment]
Some things already mentioned on slack:
- [ ] a small overview/manual of the current parsing setup: how can you start it, how to run it partially, ... This can be reused for the thesis text
- [x] if all went well, automatic cleanup of temp files (or provide an option to do this)
- [x] automatic email to unipept`@ugent`.be after running the script, containing the log file


_[Original comment](https://github.ugent.be/unipept/unipept/pull/424#issuecomment-15048) by @bmesuere on Mon Mar 23 2015 at 13:56._

**Comment by bmesuere** [comment]
general comment:
We should include a headers.txt file where the headers for each of the files are listed

comments on the assemblies file:
- only include assemblies that contain "latest" in current genbank_assembly_accession field
- field 1: is this a self assigned id?
- field 2: genbank assembly accession? same as field 6?
- field 6: may be removed
- field 9: drop the part between brackets

suggested order:
- id
- genbank_assembly_accession
- refseq_assembly_accession
- taxon_id
- genome_representation
- assembly_level
- assembly_name
- organism_name

comments on genome_sequences files
- rename to assembly_sequences
- only include assemblies that were included in the assemblies file (latest)
- drop most of the fields except:
  - id (auto increment, to be added)
  - assembly_id (the auto generated id from the assemblies file)
  - genbank_accession (without the part after the "." for example CM000663.2 should be CM000663)
  - type (for example chromosome)


_[Original comment](https://github.ugent.be/unipept/unipept/pull/424#issuecomment-15464) by @bmesuere on Wed Apr 01 2015 at 15:46._

**Pull request merged**